exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SQLExpressionBuilder.kt

```diff
@@ -339,7 +339,7 @@ interface ISqlExpressionBuilder {
         val table = (columnType as EntityIDColumnType<*>).idColumn.table as IdTable<T>
         val entityID = EntityID(t, table)
         return if ((this as? Column<*>)?.isEntityIdentifier() == true) {
-            table.mapIdComparison(entityID, ::EqOp)
+            (this as Column<*>).table.mapIdComparison(entityID, ::EqOp)
```

Comment:
Minor: cast to `Column<*>` should not be required, because it's already check in the condition. The same for `NeqOp`

Reply:
Changed. Used `this` directly to make sure local property `table` isn't used.